tests/errors/semantic/blocking/INTERFACE_CLASS__INIT__.py

```diff
@@ -0,0 +1,5 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring, missing-class-docstring
+
+class MyClass:
+    def __init__(self : 'MyClass', param1 : 'int|float'):
+        print(param1)
```

Comment:
Why should this raise an error? The input parameter is only printed out and not stored in the object, so potentially we could have multiple constructor functions which are not in contradiction with each other.

Reply:
This is true, I hadn't thought about the case where an argument is not used when @sboof911 asked me about this. I don't know how easy/difficult it is to manage Interface constructors. If it is easy then it should probably be fixed in this PR, but if not then we should probably open a dedicated issue

Reply:
@sboof911 What do you think?

Reply:
I suggest we open a dedicated issue and merge this PR now.